fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -129,7 +129,9 @@ public void processCreateMaterializedView(CreateMaterializedViewStmt stmt)
         // check db quota
         db.checkQuota();
 
-        db.writeLock();
+        if (!db.writeLockAndExist()) {
```

Comment:
My initial suggestion is to change the behavior of `db.writeLock()` and throw an exception when calling `db.writeLock()` on a dropped database. All the callers can handle this exception as it wishes.